positron/modules/process.js

```diff
@@ -16,8 +16,15 @@ Cu.import('resource://gre/modules/Services.jsm');
 
 const exeFile = Services.dirsvc.get("XREExeF", Ci.nsIFile);
 
-// Re-export `process` as a native module.  I dunno why this matters, since all
-// modules already have access to the global.  But Node does it, so we do too.
+// This file is a special kind of module, as the module loader requires it
+// during loader construction and uses it to define the `process` global.
+// Thus that global is not available at evaluation time, neither for this module
+// nor for any other modules it requires upon evaluation.
+
+const EventEmitter = require('events').EventEmitter;
+const process = new EventEmitter();
```

Comment:
> Q: shouldn't process be an object that has EventEmitter as its prototype?

The version of Node I'm integrating (5.x) [recommends either](https://nodejs.org/docs/latest-v5.x/api/events.html) the "traditional Node.js style prototypical inheritance using the util.inherits() method" or ES6 classes and *extends* to create a class that extends *EventEmitter*.

But the version of Electron I'm integrating uses either `var object = new EventEmitter` or `object.__proto__ = EventEmitter.prototype` to either create *EventEmitter* instances or decorate instances with *EventEmitter* properties and methods, depending on whether or not the object in question exists yet.

In the latter case, and in the case of `Object.create(EventEmitter.prototype)`, the object doesn't get initialized by the *EventEmitter* constructor, so its *_events* and *_eventsCount* properties are `undefined`. (Other initialization also doesn't take place, like for the *domain* property, although I'm unfamiliar with that property and don't know if this matters for Electron.)

Ironically, Node appears to anticipate *EventEmitter* being used that way, since its methods always check if *_events* is defined before using that property and *_eventsCount*. Which is why the current implementation, which uses the `object.__proto__ = EventEmitter.prototype` syntax to make *process* an *EventEmitter*, works as expected.

But in this branch I confirmed that I can `require('events')` at the top of process.js, so I figured it would make more sense to instantiate *process* as an *EventEmitter* to begin with, instead of assigning its *\_\_proto\_\_* to the *EventEmitter* prototype, to avoid the perf penalty of modifying the prototype of an object after instantiation. Hence `const process = new EventEmitter()`.